#!/bin/sh -e
## 51_gnus-pers.dpatch by Peter S Galbraith <psg@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure we're in mail header (See bug #240212)

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/gnus-bonus-el/gnus-pers.el emacs-goodies-el/elisp/gnus-bonus-el/gnus-pers.el
--- /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/gnus-bonus-el/gnus-pers.el	2004-08-16 20:49:16.000000000 -0400
+++ emacs-goodies-el/elisp/gnus-bonus-el/gnus-pers.el	2004-08-16 20:51:59.000000000 -0400
@@ -474,6 +474,7 @@
 					;Let's do From: first.
 	  (if (and from (not (equal from "")))
 	      (progn
+                (goto-char (point-min)) ;Make sure we're in the header
 		(let ((endpos (search-forward mail-header-separator)))
 		  (goto-char (point-min))
 		  (if (re-search-forward "^From:" endpos t)
